uploadNeuroDB/minc_deletion.pl

```diff
@@ -0,0 +1,286 @@
+  $selORdel  = "SELECT * ";
+}
+
+my $data_dir         = $Settings::data_dir;
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+
+sub selORdel {
+  my ($table, $field) = @_;
+  my $where = '';
+
+  if ($seriesuid) {
+    if ($table eq "parameter_file") {
+      $where = " WHERE FileID = (SELECT FileID FROM files WHERE SeriesUID = ?)";
+    } else {
+      $where = " WHERE SeriesUID = ?";
+    }
+  } elsif ($fileid) {
+    $where = " WHERE FileID =  ?";
+  }
+
+
+  my $query = $selORdel . "FROM " . $table . $where;
+  print $query . "\n";
+  my $sth = $dbh->prepare($query);
+  $sth->execute($seriesuid);
+
+  if ($selORdel eq "SELECT * ") {
+    while (my $pf = $sth->fetchrow_hashref()) {
+        print "\n$field: " . $pf->{$field};
+    }
+  }
+}
+
+
+# Delete from FS:
+# get the file names
+$query = "select f.File, f.TarchiveSource, f.SessionID, pf.`VALUE` from files as f ".
+         "left join parameter_file as pf using (FileID) where ".
+         "pf.ParameterTypeID = (select pt.ParameterTypeID from parameter_type as pt where pt.Name = 'check_pic_filename') and ".
+         "pf.FileID = ";
+
+# Useful database tracing for troubleshooting
+# $dbh->trace(5);
+
+if ($seriesuid) {
+  $query .= "(SELECT FileID FROM files WHERE SeriesUID = ?)";
+  $sth = $dbh->prepare($query);
+  $rvl = $sth->execute($seriesuid);
+} elsif ($fileid) {
+  $query .= "?";
+  $sth = $dbh->prepare($query);
+  $rvl = $sth->execute($fileid);
+}
+
+if ($sth->err) {
+  die "ERROR! return code:" . $sth->err . " error msg: " . $sth->errstr . "\n";
+}
+
+if (defined $rvl && $rvl == 0) {
+  die "Can't rename if there is no value return from: \n" . $query . "\n";
+}
+
+my $f = $sth->fetchrow_hashref();
+
+my $tarchiveid   = $f->{'TarchiveSource'};
+my $sessionid    = $f->{'SessionID'};
+my @pic_path     = split /_check/, $f->{'VALUE'};
+my $jiv_header   = $pic_path[0] . ".header";
+my $jiv_raw_byte = $pic_path[0] . ".raw_byte.gz";
+my($file, $dir, $ext) = fileparse($f->{'File'});
+my @candid = split("/", $dir);
+
+# Let's make directories
+make_path($data_dir . "/archive/"    . $dir) unless(-d  $data_dir . "/archive/"     . $dir);
+make_path($data_dir . "/archive/pic/" . $candid[1]) unless(-d  $data_dir . "/archive/pic/" . $candid[1]);
+make_path($data_dir . "/archive/jiv/" . $candid[1]) unless(-d  $data_dir . "/archive/jiv/" . $candid[1]);
+
+if ($ARGV[0] eq "confirm") {
+  rename($data_dir . "/" . $f->{'File'}, $data_dir . "/archive/" . $f->{'File'});
+  rename($data_dir . "/pic/" . $f->{'VALUE'}, $data_dir . "/archive/pic/" . $f->{'VALUE'});
+  rename($data_dir . "/jiv/" . $jiv_header, $data_dir . "/archive/jiv/" . $jiv_header);
+  rename($data_dir . "/jiv/" . $jiv_raw_byte, $data_dir . "/archive/jiv/" . $jiv_raw_byte);
+}
+
+print "Moving these files to archive:\n";
+print $data_dir . "/" . $f->{'File'} . "\n";
+print $data_dir . "/pic/" . $f->{'VALUE'} . "\n";
+print $data_dir . "/jiv/" . $jiv_header . "\n";
+print $data_dir . "/jiv/" . $jiv_raw_byte . "\n";
```

Comment:
OK then, it needs to be clearer.

if select {
   print "Files that will be moved when rerunning the script using the confirm option"
} else {
   print "Moving these files to archive"
}

Otherwise, you wonder why the files are being moved when running the script with the select option